### PR TITLE
Import utilities: Last version and mod generator

### PR DESCRIPTION
**Pull request by simpledemonn**
This pull request adds three functionalities:
- `/api/v1/latest-version-components`: Describe the latest version of all the Quilt components for all supported Minecraft versions, ready to paste into `libs.versions.toml`
- Last version page: Compose the above data into a ready to paste block, replacing for the most part LambdAurora's import utility
- Mod generator: Generator capable of taking the current template mod and customizing it according to user selections

The mod generator might be the one requiring the most testing to flush out any edge case - although I don't particularly care if it breaks on some unrealistic inputs. The generator could definitely have many more functionalities later down the line, such as integrating the Yumi license header plugin, removing the default mixin, adding more default implementation, etc..

---
See preview on Cloudflare Pages: https://preview-202.quiltmc-org.pages.dev

### CODE REVIEW

File: src/layouts/Generator.astro [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+            <div class="message-body">
+                <p>Javascript is required to view this page</p>
+            </div>
+        </article>
+    </noscript>
+
+    <p>
+        This page provides the latest versions of Quilt and its components, ready to copy paste into your <code>libs.versions.toml</code> file.
+        It is located in the <code>gradle</code> folder if you are using our template mod.
+    </p>
```
Comment (LostLuma) [inline src/layouts/Generator.astro:18]: This text should probably be different than the one on the latest versions page?

Comment (simpledemonn) [reply src/layouts/Generator.astro:18]: Fixed

**Review by LambdAurora — APPROVED** [review]

**Comment by OroArmor** [comment]
Add a way to type in a version to the latest versions page. Scrolling to find old versions can be annoying

**Comment by Pyrofab** [comment]
Unfortunately I am not aware of any straightforward solution to combine a search and a select in our current stack, so it would most likely require either a new dependency or some non-trivial development to implement. If we are going for the former, a quick search brought up [Tom Select](https://tom-select.js.org/docs/) which seems quite straightforward.

**Comment by simpledemonn** [comment]
Search field has been added

**Comment by Pyrofab** [comment]
Feels a bit awkward to use with the search field being separate yet linked to the select, but functional enough

File: src/layouts/LatestVersions.astro [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    <button id="copy-button" class="button">
+        <span class="icon">
+            <i class="fas fa-copy"></i>
+        </span>
+        <span>Copy to Clipboard</span>
+    </button>
+</Page>
+
+<script>
+    import "prismjs";
```
Comment (Pyrofab) [inline src/layouts/LatestVersions.astro:54]: Could we use the same PrismJS configuration as [the developer wiki](https://github.com/QuiltMC/developer-wiki/blob/2c9728db1adf9ffab626f36adabc704f76a150d8/src/lib/HtmlHead.svelte), for consistency and light mode support ?

**Review by Pyrofab — CHANGES_REQUESTED** [review]

File: src/layouts/LatestVersions.astro (hunk 2 above)
Comment (simpledemonn) [reply src/layouts/LatestVersions.astro:54]: Light support was added in 0707dd4

**Pull request merged**